### PR TITLE
Substitution with fuel

### PR DESCRIPTION
**Pull request by oflatt**

### CODE REVIEW

File: dag_in_context/src/utility/subst.egg [hunk 1]
```diff
@@ -52,58 +65,58 @@
...
          :ruleset subst)
-(rewrite (Subst assum to (Get c1 index))
-         (Get (Subst assum to c1) index)
+(rewrite (Subst fuel assum to (Get c1 index))
+         (Get (Subst (- fuel 1) assum to c1) index)
          :ruleset subst)
-(rewrite (Subst assum to (Alloc id c1 state ty))
-         (Alloc id (Subst assum to c1) (Subst assum to state) ty)
+(rewrite (Subst fuel assum to (Alloc id c1 state ty))
+         (Alloc id (Subst (- fuel 1) assum to c1) (Subst fuel assum to state) ty)
```
Comment (kirstenmg) [inline dag_in_context/src/utility/subst.egg:79]: Should decrease both fuels

File: dag_in_context/src/utility/subst.egg
Comment (kirstenmg) [inline dag_in_context/src/utility/subst.egg:1]: Shouldn't we stop substituting when the fuel gets to 0? I don't see where that happens in these rules.

**Review by kirstenmg — CHANGES_REQUESTED** [review]

File: dag_in_context/src/utility/subst.egg
Comment (kirstenmg) [reply dag_in_context/src/utility/subst.egg:1]: Related to this, I think there should be a test for a non-infinite fuel, e.g. a fuel of 2

**Comment by oflatt** [comment]
Don't forget to request a review when this is ready

**Pull request merged**